File: RepositoryNotifier/DTO/NotificationTO.cs
using System.Collections.Generic;
using RepositoryNotifier.TaskScheduler;

namespace RepositoryNotifier.DTO
{
    public class AddNotificationTO
    {
        public string Username { get; set; }
        public string Email { get; set; }
        public IList<string> Repositories { get; set; }
        public Frequency Frequency { get; set; }
        public IList<string> SearchKeywords{get;set;}
    }

    public class UpdateNotificationTO{
        public string Username { get; set; }
        public string Email { get; set; }
        public IList<string> Repositories { get; set; }
        public Frequency Frequency { get; set; }
        public IList<string> SearchKeywords{get;set;}
    }
}

File: RepositoryNotifier/ClientApp/src/app/dto/notificationTO.ts
export class AddNotificationTO{
    username: string;
    email: string;
    repositories: string[];
    frequency: number;
    searchKeywords: string[];
}

export class UpdateNotificationTO{
    username: string;
    email: string;
    repositories: string[];
    frequency: number;
    searchKeywords: string[];
}

export class GetNotificationTO extends AddNotificationTO{
    id: number;
    createdAt: Date;
    updatedAt: Date;
    lastExecutedAt: Date;
    status: string;
}

export class Notification{
    getNotificationTO: GetNotificationTO;
    selected: boolean;


    constructor(getNotificationTO: GetNotificationTO, selected: boolean) {
        this.getNotificationTO = getNotificationTO;
        this.selected = selected;
    }
}

export class DeleteNotificationTO{
    email: string;
    frequency: number;
}

export class NotificationResultTO{
    name:string;
    path:string;
    url:string;
    repositoryName:string;
    createdAt: Date;
}
